src/components/accession2/Accession2PlantSiteDetails.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react';
+import strings from 'src/strings';
+import { Link, Grid, Box, useTheme } from '@mui/material';
+import { AccessionPostRequestBody } from 'src/api/accessions2/accession';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+import Textfield from 'src/components/common/Textfield/Textfield';
+import { Select } from '@terraware/web-components';
+import { ACCESSION_2_COLLECTION_SOURCES } from 'src/types/Accession';
+
+type Accession2PlantSiteDetailsProps = {
+  record: AccessionPostRequestBody;
+  onChange: (id: string, value?: any) => void;
+};
+
+export default function Accession2PlantSiteDetails(props: Accession2PlantSiteDetailsProps): JSX.Element {
+  const { record, onChange } = props;
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [numPlants, setNumPlants] = useState<string>('');
+  const { isMobile } = useDeviceInfo();
+  const theme = useTheme();
+
+  const gridSize = () => (isMobile ? 12 : 6);
+
+  const onChangeNumPlants = (value: any) => {
+    const numPlantsStr = value as string;
+    setNumPlants(numPlantsStr);
+    // tslint:disable-next-line: no-unnecessary-initializer
+    let min = undefined;
+    // tslint:disable-next-line: no-unnecessary-initializer
+    let max = undefined;
+    // we support either a number or a range of two numbers, eg. 2 - 5
+    const numbers = value.split('-').map((num: string) => num.trim());
+    if (numbers.length < 3) {
+      // Parse the numbers
+      min = parseInt(numbers[0], 10);
+      max = parseInt(numbers[1], 10);
+      const noMin = isNaN(min) || min === 0;
+      const noMax = isNaN(max) || max === 0;
+      if (noMin) {
+        min = undefined;
+      }
+      if (noMax) {
+        max = min;
+      } else if (noMin) {
+        // if we have " - 3" use 3 as min and max
+        min = max;
+      } else {
+        // if we have "4 - 3" use 3 as min and 4 as max;
+        // if we have "3 - 4" use 3 as min and 4 as max;
+        const temp = min as number;
+        min = Math.min(temp, max);
+        max = Math.max(temp, max);
+      }
+    }
+    onChange('plantsCollectedFromMin', min);
```

Comment:
Values are not being saved correctly, because of this. I tested 1-10, and it shows 10-10. You have to do it like this:
`setRecord(...record, plantsCollectedFromMin: min, plantsCollectedFromMax: max` 

Because onChange is asynchronous and one call overrides the other

Reply:
hmm.. onChange just sets the property yes? it doesn't save the record until we click Add.
The BE is currently unfinished, it only saves the max property - Steve has a TODO for himself.

Reply:
Can you look in the network tab to see payload going to backend?

Reply:
In the payload I see: "plantsCollectedFromMax":10,"plantsCollectedFromMin":10

Reply:
using setRecord now